db/types.go

```diff
@@ -0,0 +1,124 @@
+package db
+
+type DB interface {
+
+	// Get returns nil iff key doesn't exist. Panics on nil key.
+	Get([]byte) []byte
+
+	// Has checks if a key exists. Panics on nil key.
+	Has(key []byte) bool
+
+	// Set sets the key. Panics on nil key.
+	Set([]byte, []byte)
+	SetSync([]byte, []byte)
+
+	// Delete deletes the key. Panics on nil key.
+	Delete([]byte)
+	DeleteSync([]byte)
+
+	// Iterator over a domain of keys in ascending order. End is exclusive.
+	// Start must be less than end, or the Iterator is invalid.
+	// CONTRACT: No writes may happen within a domain while an iterator exists over it.
+	Iterator(start, end []byte) Iterator
+
+	// Iterator over a domain of keys in descending order. End is exclusive.
+	// Start must be greater than end, or the Iterator is invalid.
+	// CONTRACT: No writes may happen within a domain while an iterator exists over it.
+	ReverseIterator(start, end []byte) Iterator
+
+	// Releases the connection.
+	Close()
+
+	// Creates a batch for atomic updates.
+	NewBatch() Batch
+
+	// For debugging
+	Print()
+
+	// Stats returns a map of property values for all keys and the size of the cache.
+	Stats() map[string]string
+}
+
+//----------------------------------------
+// Batch
+
+type Batch interface {
+	SetDeleter
+	Write()
```

Comment:
No error?

Reply:
No error for any IO operation.  See other comment.

Reply:
Gotcha, I read your other comment, thanks.